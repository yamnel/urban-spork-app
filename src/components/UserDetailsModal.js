import React from 'react';
import {Button, Modal, ModalBody, ModalFooter, ModalHeader} from 'reactstrap';
import StaticUserDetail from './StaticUserDetail';
import EditUserDetail from "./EditUserDetail";
import UrbanSporkAPI from "../api/UrbanSporkAPI";


class UserDetailsModal extends React.Component {

    state = {
        edit: false
    };
    handleOnEdit = () => {
        this.setState({edit: true})
    };
    handleOnCancel = () => {
        if (this.state.edit) {
            this.setState({edit: false});
        } else {
            this.props.toggle();
        }
    };
    handleOnClose = () => {
        this.setState({edit: false});
        this.props.toggle();
    };
    handleOnSave = () => {
        const data = {
            ForID: this.state.userData.userId,
            FirstName: this.state.userData.firstName,
            LastName: this.state.userData.lastName,
            Email: this.state.userData.email,
            Position: this.state.userData.position,
            Department: this.state.userData.department,
            IsAdmin: this.state.userData.isAdmin
        };

        UrbanSporkAPI.updateUserDetails(data)
            .then(this.setState({edit: false}))
            .then(this.props.setUserData(this.state.userData.userId, this.state.userData))

        // UrbanSporkAPI.updateUserDetails(data)
        //     .then(this.props.setUserData(this.state.userData.userId, this.state.userData))
        //     .then(this.setState({edit: false}))

    };

    handleOnDataChange = (data) => {
        console.log('the data is ', data);
        this.setState((prevState) => ({userData: {...prevState.userData, ...data}}));
        console.log('The state after change is ', this.state)
    };

    componentWillReceiveProps(nextProps) {
        if (nextProps.isOpen) {
            const userData = UrbanSporkAPI.getUserFullData(nextProps.userId);
            userData.then(data => this.setState({userData: data}))
        }
    }

    render() {
        return (
            <div>
                <Modal isOpen={this.props.isOpen} toggle={this.handleOnClose}>
                    <ModalHeader toggle={this.handleOnClose}>User Detail</ModalHeader>
                    <ModalBody>
                        <div>
                            {
                                this.state.userData ? (
                                    this.state.edit ? <EditUserDetail
                                        onDataChange={this.handleOnDataChange}
                                        userData={this.state.userData}
                                    /> : <StaticUserDetail
                                        userData={this.state.userData}/>
                                ) : null
                            }
                        </div>
                    </ModalBody>
                    <ModalFooter>
                        {this.state.edit ?
                            <Button color="success" onClick={this.handleOnSave} active={!this.state.edit}>Save</Button>
                            :
                            <Button color="primary" onClick={this.handleOnEdit} active={!this.state.edit}>Edit</Button>
                        }{' '}
                        <Button color="secondary" onClick={this.handleOnCancel}>Cancel</Button>
                    </ModalFooter>
                </Modal>
            </div>
        );
    }
}

export default UserDetailsModal;