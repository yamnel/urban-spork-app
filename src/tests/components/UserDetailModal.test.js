import React from 'react';
import {shallow} from 'enzyme';
import UserDetailsModal from "../../components/UserDetailsModal";


test('should render UserDetailsModal component correctly with no data', () => {
    const wrapper = shallow(<UserDetailsModal/>);
    expect(wrapper).toMatchSnapshot()
});
