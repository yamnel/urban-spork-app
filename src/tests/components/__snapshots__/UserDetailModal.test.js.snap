// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserDetailsModal component correctly with no data 1`] = `
<div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      User Detail
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <div />
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        active={true}
        color="primary"
        onClick={[Function]}
        tag="button"
      >
        Edit
      </Button>
       
      <Button
        color="secondary"
        onClick={[Function]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
